x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/interceptor/BulkShardRequestInterceptor.java

```diff
@@ -48,17 +49,20 @@ public void intercept(
         ActionListener<Void> listener
     ) {
         if (requestInfo.getRequest()instanceof BulkShardRequest bulkShardRequest) {
+            final boolean isDlsLicensed = DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState);
+            final boolean isFlsLicensed = FIELD_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState);
+            if (isDlsLicensed == false && isFlsLicensed == false) {
+                listener.onResponse(null);
+                return;
+            }
```

Comment:
I understand that the intention is to only check the DLS/FLS permissions if the feature is licensed such that we avoid computing those otherwise. But this is hard to notice at this level.
What do you think if we move the check inside a `IndicesAccessControl#hasFieldLevelSecurity(String indexName)`? It should return `false` if it is not licensed, and otherwise trigger the access control computation and track the feature usage.

Reply:
I think this would cause issue in `DlsFlsLicenseRequestInterceptor`. There we check the compatibility of license and DLS/FLS usage.

Reply:
You're right, it's not a viable suggestion.

I think it's good the way you have it here, to first check if the features are licensed and only afterwards proceed to verify them (rather than the reverse in the original code).

One small nit though, could you please incorporate the `isDlsLicensed || isFlsLicensed` in the if above, I think it's easier on the eyes.